submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/FileListSubmissionTest.kt

```diff
@@ -179,34 +181,86 @@ internal class FileListSubmissionTest(private val tempFolder: TemporaryFolder) :
             val createdSub = submissionRepository.getExtByAccNo(accNo)
             val subFolder = "$submissionPath/${createdSub.relPath}"
 
-            val fileListName = createdSub.section.fileList?.fileName
-            val expectedTabFiles = if (mongoMode) {
-                if (enableFire) listOf() else listOf(
-                    NfsFile("$fileListName.json", File(subFolder).resolve("Files/$fileListName.json")),
-                    NfsFile("$fileListName.xml", File(subFolder).resolve("Files/$fileListName.xml")),
-                    NfsFile("$fileListName.pagetab.tsv", File(subFolder).resolve("Files/$fileListName.pagetab.tsv"))
-                )
-            } else listOf()
-
-            val submissionPageTabFiles = if (mongoMode) {
-                if (enableFire) listOf() else listOf(
-                    NfsFile("$accNo.json", File(subFolder).resolve("$accNo.json")),
-                    NfsFile("$accNo.xml", File(subFolder).resolve("$accNo.xml")),
-                    NfsFile("$accNo.pagetab.tsv", File(subFolder).resolve("$accNo.pagetab.tsv"))
-                )
-            } else listOf()
-
-            assertThat(createdSub.section.fileList).isEqualTo(ExtFileList("FileList", pageTabFiles = expectedTabFiles))
-            assertThat((createdSub.pageTabFiles)).isEqualTo(submissionPageTabFiles)
+            if (mongoMode)
+                if (enableFire) {
+                    val submissionTabFiles = createdSub.pageTabFiles as List<FireFile>
+                    assertThat(submissionTabFiles).hasSize(3)
+                    assertThat(submissionTabFiles).isEqualTo(submissionFireTabFiles(accNo, subFolder))
+
+                    val fileListTabFiles = createdSub.section.fileList!!.pageTabFiles as List<FireFile>
+                    assertThat(fileListTabFiles).hasSize(3)
+                    assertThat(fileListTabFiles).isEqualTo(fileListFireTabFiles(subFolder))
+                } else {
+                    val submissionTabFiles = createdSub.pageTabFiles as List<NfsFile>
+                    assertThat(submissionTabFiles).hasSize(3)
+                    assertThat(submissionTabFiles).isEqualTo(submissionNfsTabFiles(accNo, subFolder))
+
+                    val fileListTabFiles = createdSub.section.fileList!!.pageTabFiles as List<NfsFile>
+                    assertThat(fileListTabFiles).hasSize(3)
+                    assertThat(fileListTabFiles).isEqualTo(fileListNfsTabFiles(subFolder))
+                }
 
             assertThat(Paths.get("$subFolder/Files/$testFile")).exists()
-            assertThat(Paths.get("$subFolder/Files/$fileListName.xml")).exists()
-            assertThat(Paths.get("$subFolder/Files/$fileListName.json")).exists()
-            assertThat(Paths.get("$subFolder/Files/$fileListName.pagetab.tsv")).exists()
+            assertThat(Paths.get("$subFolder/Files/FileList.xml")).exists()
```

Comment:
Why are the page tab files on NFS always checked? Shouldn't this be checked only if the system is not running on fire mode?

I think we should have something like:

`if (enableFire) assertFireSubmissionFiles() else assertNfsSubmissionFiles()`

And then, move the assertion logic to this methods accordingly

Reply:
TestWireMockTransformer class store the files in the same path in that way we do not need specific logic for this part for each system (Fire and NFS).